Move condition field from options to specifications tab

jolelievre opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Move condition field from options to specifications tab And add new options to `SwitchType` to customize its display which can now match the mockups (Also turn every h2 header into h3 in the product form)
| Type?             | improvement
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | ~
| How to test?      | Change the condition in the specifications tab It should work
| Possible impacts? | ~


<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/27275)
<!-- Reviewable:end -->

### Breaking Changes

`OptionsType` and `SpecificationType` constructor parameters have changed, but they are injected via the definition of service so it shouldn't have much impact

jolelievre committed:
Move condition fields from options tab to the specification tab

jolelievre committed:
Update new format for product form data condition

jolelievre committed:
Options command builder fetch data from the new format for condition

jolelievre committed:
All label headers should be h3 not h2

jolelievre committed:
SwitchType labels are hidable

jolelievre committed:
Switch type can be displayed inline, label and switch button on the same line

jolelievre committed:
Adapt condition display to the mockup

matthieu-rolland commented on src/PrestaShopBundle/Form/Admin/Sell/Product/Specification/SpecificationsType.php:
```diff
@@ -44,6 +67,24 @@ public function buildForm(FormBuilderInterface $builder, array $options)
+                'required' => false,
+                // placeholder false is important to avoid empty option in select input despite required being false
+                'placeholder' => false,
+                'label' => false,
```
is there a reason to set the label to false ? (it had a 'condition' string when it was in  `OptionsType` )

jolelievre replied:
Yes in the new design this input no longer has a label, we only have a label on the checkbox before

jolelievre replied:
![Capture d’écran 2022-01-10 à 10 53 32](https://user-images.githubusercontent.com/13801017/148746784-7936dcf4-2b01-46ab-b192-e563186bf1d6.png)

jolelievre replied:
@matthieu-rolland is it legit with this explanation? ^^

matthieu-rolland reviewed (commented):
one small feedback, otherwise LGTM 👍

sowbiba reviewed (changes requested):
Please document BC breaks. Otherwise, good to me

khouloudbelguith reviewed (changes requested):
Hi @jolelievre,

I have only one behavior
The `help-box` is no more displayed after moving the Condition from Options Tab to Specifications Tab
**Without PR**
![image](https://user-images.githubusercontent.com/16067358/154051726-fd7b9838-3a93-46bc-85df-96b3889a3bd5.png)

**With PR**
![image](https://user-images.githubusercontent.com/16067358/154052012-f0a175ea-d04f-4981-9a2a-9e895019bb70.png)

Thanks!

jolelievre commented:
Thanks @khouloudbelguith 

in the current mockup the help box is not present anymore
<img width="758" alt="Capture d’écran 2022-02-16 à 15 48 53" src="https://user-images.githubusercontent.com/13801017/154289603-84b6f803-4c36-46c1-a9b0-6694d6616a95.png">

This needs to be approved by @TristanLDD and @Julievrz I guess, if indeed this help box is totally removed I need to clean a small part of the code (the message is assigned to the select input but since it has no label the help box can't be shown, so this is dead code that can be removed without changing the behaviour)

Julievrz commented:
In my opinion, the tooltip is redundant. The label is clear enough. I think we should get rid of it. :)

jolelievre committed:
Remove help box from condition field

jolelievre commented:
Ok thanks @Julievrz 

Then I rebased the PR because it was needed and I removed the useless code that tried to add the help box without success So it is completely removed now!

Progi1984 commented on src/PrestaShopBundle/Form/Admin/Sell/Product/Options/OptionsType.php:
```diff
@@ -27,61 +27,22 @@
 /**
  * This form class is responsible to generate the product options form.
  */
 class OptionsType extends TranslatorAwareType
```
@jolelievre Please describe this BC Break :)

jolelievre replied:
@Progi1984 please check your review bot :trollface: 
I described the BC in the description about a month ago

jolelievre replied:
<img width="844" alt="Capture d’écran 2022-02-24 à 16 12 21" src="https://user-images.githubusercontent.com/13801017/155551565-26b5d3d0-5970-4bdf-ac63-b1b2fd6337b6.png">

Progi1984 reviewed (commented):
One comment and it's approved.

khouloudbelguith commented:
Hi @jolelievre,

I have checked the spec: https://github.com/PrestaShop/prestashop-specs/blob/master/content/1.7/back-office/catalog/products/add-edit-v2.md#display-condition-on-product-page

>Display condition on product page - Switch button - When enabled, it enables Condition dropdown, otherwise it is disabled

![image](https://user-images.githubusercontent.com/16067358/157077852-5ef9d407-e273-4106-979b-2174251ee986.png)

It should be 
![image](https://user-images.githubusercontent.com/16067358/157078005-50c22a47-cef0-4484-8d15-f242ac51e081.png)

Ping @MatShir what do you think?

Thanks!

jolelievre commented:
Hi @khouloudbelguith 

indeed I'm waiting for another PR to be merged so that I can use a new JS component. I don't want to reinvent the wheel or create duplicate PRs or conflicts ^^

So I suggest opening an issue for this last detail

khouloudbelguith reviewed (approved):
Hi @jolelievre,

It is ok :heavy_check_mark: 


https://user-images.githubusercontent.com/16067358/157227994-04c020c4-7c37-4ad7-9415-d035b5e88fbd.mp4

**Spec**
- Display condition on product page - Switch button => ok
- When enabled, it enables Condition dropdown, otherwise it is disabled: enabled => ok, disabled => Nok https://github.com/PrestaShop/PrestaShop/issues/27855
- Condition: Drop-down with 3 options: New, Used, Refurbished. => ok
- By default, New is selected. => OK

--------------------------------------------------

- I checked with multishop enabled => OK
- I checked different types of products (Simple, with combinations, pack, vector) => OK
- I checked with mobile

https://user-images.githubusercontent.com/16067358/157229433-90a6f05f-12c4-43d1-8ef4-9b87f09c2f1d.mp4


>indeed I'm waiting for another PR to be merged so that I can use a new JS component. I don't want to reinvent the wheel or create duplicate PRs or conflicts ^^
So I suggest opening an issue for this last detail

I created a new issue: https://github.com/PrestaShop/PrestaShop/issues/27855

Thanks!

Progi1984 commented:
Thanks @jolelievre & @khouloudbelguith
